src/middleware/validate/validateAddressMiddleware.ts

```diff
@@ -35,11 +35,21 @@ function checkAddress(address: string): [boolean, string | undefined] {
 	try {
 		decoded = base58Decode(address);
 	} catch (error) {
+		// If ss58 doesn't pass, we try H160 address.
+		if (address.startsWith('0x')) {
+			decoded = Uint8Array.from(Buffer.from(address.slice(2), 'hex'));
```

Comment:
"decoded" here is a bit unclear. On line 36 the address is decoded from base58 to hex. Here though, "decoding" means a different thing: cast a hex string to bytes. I would prefer clearer naming to properly distinguish the kind of decoding happening. Maybe `hexAddress`?